### DISCUSSION TITLE
Git compatibility (and other collaboration tools)

### DISCUSSION THREAD

**Discussion by bitbrain** [bitbrain/pandora#64] [Proposals]
Currently, Pandora defaults to compressed json format to store its data. However, when two or more developers change some config in Pandora and commit it, the merge conflicts are unresolvable, since we suddenly have two binary files in conflict.

## Why compressing in the first place?

My original thinking was to allow compression to save disk space but that ideally should only happen when actually exporting the game (so only write/parse compressed file for production builds).

## Ideas

A. toggle compression via project property
B. auto-format json (one entry per line) for better Git conflict resolution
C. ???

Ideas welcome!

**Comment by eth0net** [comment]
I agree with the idea of compressing it in production, but during development this isn't ideal as mentioned.

If it's simple enough to do, having a standard formatted JSON file for development and compressed / minified file for release sounds perfect.

My only concern with a compressed file for release is how accessible it would be for potential mod makers. Since the file is still understood by Pandora, anyone using Godot for mod/dlc development would be fine, but is it easy enough to open for editing inside other code editors? This is purely from a point of not wanting to enforce Godot for mod making, but that's not the end of the world!

**Comment by mrfelipemartins** [comment]
In my opinion during development the data file should be uncompressed in Pretty Printed JSON Format, as you mentioned this would help a lot in Git conflict resolution. Only when exporting the game it should be compressed.

About @eth0net concerns about modding, The idea proposed by @bitbrain on https://github.com/bitbrain/pandora/discussions/80 would solve this if Pandora had an out of the box mod loader, so modders wouldn't need to mess with the compressed file.

**Reply by eth0net** [reply]
This all seems fair ^^

I can 100% see someone complaining they can't open it in VS Code or whatever, but if they can handle it in Godot then not much they have to really complain about ^^ 

I'd be onboard with dev being pretty JSON and prod being compressed, with the option to change either. The work I'm doing on ID regeneration includes a settings view so that can be reused for this!

**Reply by mrfelipemartins** [reply]
Take a look at #106!

I am down to implement this as an option if needed.

**Reply by eth0net** [reply]
Once #103 is merged the settings will be in place, so you can look at adding it then if you want 🙂
